src/main/scala/net/snowflake/spark/snowflake/pushdowns/querygeneration/WindowStatement.scala

```diff
@@ -46,6 +47,14 @@ private[querygeneration] object WindowStatement {
               useWindowFrame = true
             )
 
+          // Disable window function pushdown if
+          // 1. The function are both window function and aggregate function
+          // 2. The function support Window Frame
+          // 3. The user specify ORDER-BY clause
+          case agg: AggregateExpression
+            if AggregationStatement.supportWindowFrame(agg) && spec.orderSpec.nonEmpty =>
```

Comment:
Why can;t you check if there is in fact a window function ? Isn;t that easier ?

Reply:
I still don't understand why we care about `supportWindowFrame`. If there is a window frame in Spark, even though Snowflake may not support it, why do we want to default to the `case _` behavior? Shouldn't we just bail out with `null` in that case?